cmake/AwsCheckHeaders.cmake

```diff
@@ -40,20 +40,29 @@ function(aws_check_headers target)
             LINKER_LANGUAGE CXX
             CXX_STANDARD 11
             CXX_STANDARD_REQUIRED 0
-            C_STANDARD 90
+            C_STANDARD 99
```

Comment:
If these are for cpp c99 is too modern right?

Reply:
When it's C_STANDARD 90 and warnings are pedantic, we get slammed with stuff like "variadic macros are a C99 feature" and "bool is a c99 feature"

Remember how we separately generate a .cpp file and a .c file? So we stress including the header from C and we stress including it from C++. The .cpp files are passing so it seems ok?

Reply:
I forgot that the simple act of using a c++ compiler will be sufficient to check those headers. So I'm fine with this.